homeassistant/components/smartthings/binary_sensor.py

```diff
@@ -69,25 +77,32 @@ def get_capabilities(capabilities: Sequence[str]) -> Sequence[str] | None:
 class SmartThingsBinarySensor(SmartThingsEntity, BinarySensorEntity):
     """Define a SmartThings Binary Sensor."""
 
-    def __init__(self, device, attribute):
+    def __init__(self, device, component, attribute):
         """Init the class."""
         super().__init__(device)
+        self._component = component
         self._attribute = attribute
 
     @property
     def name(self) -> str:
         """Return the name of the binary sensor."""
-        return f"{self._device.label} {self._attribute}"
+        if self._component == "main":
+            return f"{self._device.label} {self._attribute}"
+        return f"{self._device.label} {self._component} {self._attribute}"
 
     @property
     def unique_id(self) -> str:
         """Return a unique ID."""
-        return f"{self._device.device_id}.{self._attribute}"
+        if self._component == "main":
+            return f"{self._device.device_id}.{self._attribute}"
+        return f"{self._device.device_id}.{self._component}.{self._attribute}"
 
     @property
     def is_on(self):
         """Return true if the binary sensor is on."""
-        return self._device.status.is_on(self._attribute)
+        if self._component == "main":
```

Comment:
Is this conditional line needed?

Reply:
I think status data for main is only on the primary status object, not nested in the component status objects, therefore we need to branch every time we try to fetch the current status.

Alternately, we could modify the base library further to refactor how the main component stores it's statuses, but that is a breaking change.